node/test/Test/Pos/Client/Txp/UtilSpec.hs

```diff
@@ -0,0 +1,246 @@
+    manyUtxoTo1Desc =
+        "Transaction is created successfully when we have 10 items in Utxo " <>
+        "for a single address with 100k coins each"
+    manyAddressesTo1Desc =
+        "Transaction is created successfully when we have 10 items in Utxo " <>
+        "for 10 different addresses with 100k coins each and 1 output with 1 coin"
+    manyAddressesToManyDesc =
+        "Transaction is created successfully when we have 10 items in Utxo " <>
+        "for 10 different addresses with 100k coins each and 10 outputs with 1 coin each"
+    redemptionDesc =
+        "Redemption transaction is created successfully"
+    txWithRedeemOutputFailsDesc =
+        "An attempt to create a tx with a redeem address as an output fails"
+
+createMTxWorksWhenWeAreRichSpec :: HasCoreConstants => TxpTestProperty ()
+createMTxWorksWhenWeAreRichSpec = do
+    txOrError <- createMTx cmpUtxo cmpSigners cmpOutputs cmpAddrData
+    case txOrError of
+        Left err -> stopProperty $ sformat ("Failed to create tx: "%build) err
+        Right tx -> ensureTxMakesSense tx cmpUtxo cmpOutputs
+  where
+    CreateMTxParams {..} = makeManyAddressesToManyParams 1 1000000 1 1
+
+stabilizationDoesNotFailSpec :: HasCoreConstants => TxpTestProperty ()
+stabilizationDoesNotFailSpec = do
+    txOrError <- createMTx cmpUtxo cmpSigners cmpOutputs cmpAddrData
+    case txOrError of
+        Left err@(FailedToStabilize _) -> stopProperty $ pretty err
+        Left _                         -> return ()
+        Right tx                       -> ensureTxMakesSense tx cmpUtxo cmpOutputs
+  where
+    CreateMTxParams {..} = makeManyAddressesToManyParams 1 200000 1 1
+
+feeIsNonzeroSpec :: HasCoreConstants => TxpTestProperty ()
+feeIsNonzeroSpec = do
+    txOrError <- createMTx cmpUtxo cmpSigners cmpOutputs cmpAddrData
+    case txOrError of
+        Left (NotEnoughMoney _) -> return ()
+        Left err -> stopProperty $ pretty err
+        Right _ -> stopProperty $
+            sformat ("Transaction was created even though there were "%
+                "not enough funds for the fee")
+  where
+    CreateMTxParams {..} = makeManyAddressesToManyParams 1 100000 1 1
+
+manyUtxoTo1Spec :: HasCoreConstants => TxpTestProperty ()
+manyUtxoTo1Spec = do
+    txOrError <- createMTx cmpUtxo cmpSigners cmpOutputs cmpAddrData
+    case txOrError of
+        Left err -> stopProperty $ pretty err
+        Right tx -> ensureTxMakesSense tx cmpUtxo cmpOutputs
+  where
+    CreateMTxParams {..} = makeManyUtxoTo1Params 10 100000 1
+
+manyAddressesTo1Spec :: HasCoreConstants => TxpTestProperty ()
+manyAddressesTo1Spec = do
+    txOrError <- createMTx cmpUtxo cmpSigners cmpOutputs cmpAddrData
+    case txOrError of
+        Left err -> stopProperty $ pretty err
+        Right tx -> ensureTxMakesSense tx cmpUtxo cmpOutputs
+  where
+    CreateMTxParams {..} = makeManyAddressesToManyParams 10 100000 1 1
+
+manyAddressesToManySpec :: HasCoreConstants => TxpTestProperty ()
```

Comment:
Can be merged with `manyAddressesTo1Spec`.

Reply:
Not sure how. Can you elaborate?

Reply:
Two functions _only_ differ in the third argument of `makeManyAddressesToManyParams`.